File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int number)
{
    int size = trunc(number <= 0 ? 0 : log10(number)) + 1;
    int sum = 0;
    for (int n = number; n > 0; n /= 10)
    {
        sum += pow(n % 10, size);
    }
    return number == sum;
}

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

func IsNumber(number int) bool {
	var sum, size int
	for n := number; n != 0; n /= 10 {
		size++
	}
	for n := number; n != 0; n /= 10 {
		sum += pow(n%10, size)
	}
	return sum == number
}

func pow(n, e int) int {
	if e == 0 {
		return 1
	}
	result := n
	for i := 2; i <= e; i++ {
		result *= n
	}
	return result
}
